lib/matplotlib/widgets.py

```diff
@@ -1417,6 +1417,543 @@ def _update(self):
+        # Else the ruler follow the mouse cursor
+        else:
+            pos_a = self.x0, self.x1
+            pos_b = self.y0, self.y1
+
+        self.ruler.set_data([pos_a], [pos_b])
+        x1 = self.ruler.get_path().vertices[1][0]
+        y1 = self.ruler.get_path().vertices[1][1]
+
+        self.marker_c.set_visible(True)
+        self.marker_c.set_data(x1, y1)
+
+        self.set_midline_marker()
+        self.update_text()
+        self._update_artists()
+
+    @property
+    def line_coords(self):
+        line_coords = self.ruler.get_path().vertices
+        x0 = line_coords[0][0]
+        y0 = line_coords[0][1]
+        x1 = line_coords[1][0]
+        y1 = line_coords[1][1]
+
+        pos_a = x0, y0
+        pos_b = x1, y1
+
+        return pos_a, pos_b
+
+    def _update_artists(self):
+        if self.useblit:
+            if self.background is not None:
+                self.canvas.restore_region(self.background)
+
+            for artist in self.artists:
+                self.fig.draw_artist(artist)
+
+            self.fig.canvas.blit(self.fig.bbox)
+        else:
+            self.canvas.draw_idle()
+
+    def update_text(self):
+        if self.length_unit is not None:
+
+            detail_string = 'L: {:0.3f} {}; dx: {:0.3f} {}; dy: {:0.3f} {}; ' \
+                            'angle: {:0.3f} deg'.format(self.ruler_length,
+                                                        self.length_unit,
+                                                        self.ruler_dx,
+                                                        self.length_unit,
+                                                        self.ruler_dy,
+                                                        self.length_unit,
+                                                        self.ruler_angle)
+        else:
+            detail_string = 'L: {:0.3f}; dx: {:0.3f}; dy: {:0.3f}; ' \
+                            'ang: {:0.3f} deg'.format(self.ruler_length,
+                                                      self.ruler_dx,
+                                                      self.ruler_dy,
+                                                      self.ruler_angle)
+
+        self.axes_text.set_text(detail_string)
+        if self.print_text is True:
+            print(detail_string)
+
+    def on_release(self, event):
+        self.mouse1_pressed = False
+        self.mouse3_pressed = False
+        self.ruler_moving = False
+        self.end_a_lock = False
+        self.end_b_lock = False
+        self.end_c_lock = False
+
+        if event.inaxes != self.ax.axes:
+            return
+
+        if self.useblit:
+            for artist in self.artists:
+                artist.set_animated(False)
+
+        self.canvas.draw_idle()
+
+    @property
+    def ruler_length(self):
+        pos0, pos1 = self.line_coords
+
+        return np.sqrt((pos1[0] - pos0[0]) ** 2 + (pos0[1] - pos1[1]) ** 2)
+
+    @property
+    def ruler_dx(self):
```

Comment:
I guess in my use case I only care about the absolute value. But I can see your reasoning. It would probably be helpful to most if the correct sign is return.